### PR TITLE
server/server.rs: Fix http_proxy brokes test_peer_resolve

### PR DESCRIPTION
**Pull request by MyonKeminta**
Remove environment variable `http_proxy` and `https_proxy` before running test `test_peer_resolve`. So we don't need to unset them before running the test.
Previously `http_proxy` will broke this test.

### CODE REVIEW

**Comment by BusyJay** [comment]
Interesting. Can this be resolved by changing the configuration of channel instead of removing environment variables?

**Comment by siddontang** [comment]
can we remove this at first before running all test?

**Comment by MyonKeminta** [comment]
@siddontang Did you mean removing them in makefile? The cargo test command may build the project first, which might fetch crates from internet.

**Comment by siddontang** [comment]
Got it.

Maybe we can do it in `_0_ci_setup`

**Comment by MyonKeminta** [comment]
@siddontang But it doesn't solve the trouble that I will fail `test_peer_resolve` when I run `make dev` with `http_proxy` set on my computer.
I think @BusyJay 's suggestion is the best approach if it's possible.

**Comment by huachaohuang** [comment]
So, how's this go now?

**Comment by siddontang** [comment]
Ping @MyonKeminta

File: src/server/server.rs [hunk 1]
```diff
@@ -260,8 +261,9 @@ mod tests {
     }
 
     #[test]
-    // if this failed, unset the environmental variables 'http_proxy' and 'https_proxy', and retry.
     fn test_peer_resolve() {
+        env::remove_var("http_proxy");
```
Comment (siddontang) [inline src/server/server.rs]: add a detailed comment why we need doing here.

Rest LGTM

**Comment by breeswish** [comment]
This solution seems to be not elegant. It only fixes a specific issue and is not generic enough to prevent similar issues. I prefer jay's idea. We can have a global config to configure whether or not http(s)_proxy should be respected. And in test environment it should be false.

**Comment by BusyJay** [comment]
Looks like you can add `172.0.0.1` and `localhost` to `no_proxy` environment variable to fix it.

**Comment by siddontang** [comment]
any update? @MyonKeminta

**Comment by overvenus** [comment]
What about checking envs in `_1_check_system_requirement`?

**Comment by huachaohuang** [comment]
Seems this PR is not important and has not been updated for a long time, let's close it.

**Pull request closed without merging**